myhoard/web_server.py

```diff
@@ -73,10 +73,9 @@ async def backup_list(self, _request):
             response = {
                 "backups": None,
             }
-            with self.controller.lock:
```

Comment:
The change for `controller` is valid but why is the lock removed here? We need both for the access to be properly protected.

Reply:
mmm I think I am a bit confused since we do not apply acquire the lock when accessing to backups, so not sure why we should do here. Also, maybe we should avoid accessing to the controller's state from other places?  Is it better to have `Controller.get_backups()` and lock it from there?

Reply:
So that both the `backups_fetched_at` and `backups` are accessed in a consistent state. Otherwise a statement that updates these values could've already set `backups_fetched_at` to some value and then we assume here backups are available but then that value hasn't yet been set so we could end up returning empty backup list instead of `null`.

Controller itself doesn't need the lock because it is the only one that is making updates to the backups and the main logic there runs on the same thread so there cannot be concurrent update.

Reply:
ah fair enough, thanks for the explanation :+1:  will bring it back